packages/raja/package.py

```diff
@@ -195,6 +208,7 @@ class Raja(CachedCMakePackage, CudaPackage, ROCmPackage):
     depends_on("camp+openmp", when="+openmp")
     depends_on("camp@main", when="@develop")
     depends_on("camp@main", when="@main")
+    depends_on("camp@2024.02.1:", type="build", when="@2024.02.1:")
```

Comment:
@adrienbernede can we get the changes in the other PR pulled into this one: https://github.com/LLNL/radiuss-spack-configs/pull/106

The references to branch dependencies break our CI due to the way Spack does mirroring. @white238 can explain the details better than me. The other PR fixes the issue.

We are preparing new releases for RAJA, Umpire, and CHAI and will have new version info for those.

Reply:
I suspect the usage of branches in combination of a source mirror was causing Spack to lock down the git branch mirror to an old commit that wasn't updating. I have noticed this behavior before when you add a package to a spack mirror that uses a branch, Spack doesn't track the commit moved forward, just that the branch exists in the repo and will not update it. So @rhornung67  was getting and old camp.

Reply:
But... @white238 I don’t remember using a source mirror... does it apply to binary caches too ?

Reply:
My guess is it applies to both. It has to do with the fact that Spack treats branch names as versions. So the branch name gets locked down at a first come first serve basis.